pkgs/top-level/stage.nix

```diff
@@ -176,62 +176,80 @@ let
       ((config.packageOverrides or (super: {})) super);
 
   # Convenience attributes for instantitating package sets. Each of
-  # these will instantiate a new version of allPackages. Currently the
-  # following package sets are provided:
-  #
-  # - pkgsCross.<system> where system is a member of lib.systems.examples
-  # - pkgsMusl
-  # - pkgsi686Linux
-  otherPackageSets = self: super: {
+  # these will instantiate a new version of allPackages.
+  otherPackageSets = let
+    # helper function to call nixpkgsFun to make sure each package set
+    # references itself as a no-op and to prevent nested pkgsCross.
+    # Nested pkgsCross doesn't make sense fundamentally, because it replaces
+    # the full crossSystem argument, thus starting "from scratch".
+    createPackageSet = name: { overlays ? [], ... }@args:
+      nixpkgsFun (args // {
+        overlays = [
+          (self: super: {
+            "${name}" = super;
+            pkgsCross = throw "Nested pkgsCross not allowed.";
```

Comment:
Why not allow this if it is well-defined semantically? Otherwise, it would be a breaking change:

```
❯ nom-build default.nix -A pkgsCross.mingwW64.hello
Finished at 13:39:42 after 0s
/nix/store/7k2v0zk1f3aypihalihygm6bzcly8v38-hello-x86_64-w64-mingw32-2.12.1
❯ nom-build default.nix -A pkgsCross.mingwW64.pkgsCross.mingwW64.hello
Finished at 13:39:31 after 0s
/nix/store/7k2v0zk1f3aypihalihygm6bzcly8v38-hello-x86_64-w64-mingw32-2.12.1
```

Reply:
I agree. The following example should NOT produce the same hashes:

```
❯ nix-build default.nix -A pkgsCross.aarch64-multiplatform.pkgsCross.gnu64.hello
/nix/store/qdaj356l85gyahgvlwmj9g955pkpsfgv-hello-2.12.1
❯ nix-build default.nix -A hello
/nix/store/qdaj356l85gyahgvlwmj9g955pkpsfgv-hello-2.12.1
```